packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -83,11 +86,20 @@ function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
     );
   }
 
-  const devices = parseIOSDevicesList(
-    child_process.execFileSync('xcrun', ['instruments', '-s'], {
-      encoding: 'utf8',
-    }),
-  );
+  const env = await getEnvironmentInfo(true);
+  const xcodeVersion = semver.coerce(env.IDEs.Xcode.version.split('/')[0]);
+  const devices =
+    xcodeVersion != null && semver.satisfies(xcodeVersion, '>= 12.x')
+      ? parseXtraceIOSDevicesList(
+          child_process.execFileSync('xcrun', ['xctrace', 'list', 'devices'], {
+            encoding: 'utf8',
+          }),
```

Comment:
Also, while testing locally, I am getting output of this command printed in my terminal and `text` passed to `parseXtraceIOSDevicesList` is `null`.

Not sure why is that. Do we need to overwrite `stdio` here? The best scenario is when there's nothing outputted to the console. We just get the list and assign to `devices`.

We do the printing later anyways, on our own. Otherwise, we will print list of devices twice.